### PR TITLE
When finding a conflict compare sites without protocol

### PR DESCRIPTION
**Pull request by enejb**
Sometimes sites that switch from .com => jetpack cause conflicts. This PR tries to fix an edge case where the domain that we compare went from http -> https and don't mark them having a conflict. 

This PR fixed this case by making sure that we compare urls without the protocol

cc: @ebinion.

### CODE REVIEW

**Comment by ebinnion** [comment]
I tested this and can confirm the fix. Instead of having two URLs that use the mapped domain, the WP.com site ends up `$site.wordpress.com`.

As a note, in the specific case I tested, the site was moved from `https` at WP.com to `http` on the remote site.

File: client/lib/sites-list/list.js [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
+import wpcom from 'lib/wp';
+import Site from 'lib/site';
+import JetpackSite from 'lib/site/jetpack';
+import Searchable from 'lib/mixins/searchable';
+import Emitter from 'lib/mixins/emitter';
+import { isPlan } from 'lib/products-values';
+import userUtils from 'lib/user/utils';
+import { withoutHttp } from 'lib/url';
+
+
```
Comment (ebinnion) [inline client/lib/sites-list/list.js]: There's an extra blank line here.

**Comment by ebinnion** [comment]
Reviewed code and I don't have comments. The only feedback I have is that we should get some input on whether this will still work as we move from sites list to redux.

So, I'd suggest we move forward with this for now, and then we can begin working on the WPCOM side to work out these collisions or do some testing to make sure that this is also handled with the redux site list.

**Comment by enejb** [comment]
This should be handled on the redux side. See https://github.com/Automattic/wp-calypso/blob/b6d0d27438a16cf7c8700cf4ed8b70dbc42805e3/client/state/sites/selectors.js#L68

**Pull request merged**